src/organisms/subject-attend.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from 'react';
+import AttendanceStatus from '../molecules/attendance-status';
+import { API_ROOT } from "@/queries/config";
+
+export type SubjectAttendanceProps = {
+  subjectId: string;
+};
+
+const SubjectAttend = ({ subjectId }: SubjectAttendProps) => {
+    const [subject, setSubject] = useState(null);
+  
+    useEffect(() => {
+      const fetchSubjectData = async () => {
+        try {
+          const url = `${API_ROOT}/attendances/${subjectId}`;
+          const response = await fetch(url);
+          const data = await response.json();
+  
+          if (response.ok) {
+            setSubject({
+              id: subjectId,
+              name: '科目名',
+              attendances: data,
+            });
+          } else {
+            setSubject(null);
+          }
```

Comment:
```suggestion
          if (!response.ok) {
              return;
          }

          const data = (await response.json()) as AttendancesSum;
          setSubject(data);
```
レスポンスが正常なステータス (ok) かどうかを確認してから JSON データを取り出しましょう.
また, 取り出したデータの形式が仕様と合っていません.
そして, 初期値である `null` を設定する意味は無いと思います.